src/hotspot/share/runtime/synchronizer.cpp

```diff
@@ -2333,192 +1455,108 @@ class HandshakeForDeflation : public HandshakeClosure {
   }
 };
 
-// This function is called by the ServiceThread to deflate monitors.
-// It is also called by do_final_audit_and_print_stats() by the VMThread.
-void ObjectSynchronizer::deflate_idle_monitors() {
+// This function is called by the MonitorDeflationThread to deflate
+// ObjectMonitors. It is also called via do_final_audit_and_print_stats()
+// by the VMThread.
```

Comment:
Ah! I think this addresses my previous comments about a non-JavaThread doing this.

Reply:
Yup...